### ISSUE TITLE
Learner interpreter doesn't iterate over all of the training dataset.

### ISSUE THREAD

**Issue by mm-mansour** [fastai/fastai#1914]
**Describtion**

I was trying to inspect the top losses across the training dataset so I initialized an interpreter with `ds_type=DatasetType.Train` and then called the `top_losses()` method.
The length of the returned list was 704 while the length of the training set was 709.

**Full code to reproduce:**
```

from fastai.vision import *
from fastai import *


path = untar_data(URLs.MNIST_TINY)
data = (ImageList.from_folder(path)
        .split_by_folder()
        .label_from_folder()
        .databunch(bs=32)
        .normalize()
        )

learner = cnn_learner(data, models.resnet18, metrics=accuracy)
learner.fit_one_cycle(5, 3e-3)

interp = learner.interpret()
interp_train = learner.interpret(DatasetType.Train)

valid_losses = interp.top_losses()
print(len(valid_losses[0]), len(data.valid_ds.x),
      len(valid_losses[0]) == len(data.valid_ds.x))

train_losses = interp_train.top_losses()
print(len(train_losses[0]), len(data.train_ds.x),
      len(train_losses[0]) == len(data.train_ds.x))
```
**Output:**
> 699 699 True
> 704 709 False

 
**Expected behavior**
> 699 699 True
> 709 709 True

Note how this didn't happen in the case of the validation set where the `top_losses()` function returned 699 items matching the length of the validation_set.

This seems to be related to the batch size since 704 corresponds to exactly 22 batches since 709 doesn't divide by the batch size of 32. Yet again, this wasn't the case in the validation set and it returns 699 items even though 699 doesn't divide by 32.

**Installation details**

```text
=== Software === 
python        : 3.6.6
fastai        : 1.0.52.dev0
fastprogress  : 0.1.19
torch         : 1.0.1.post2
nvidia driver : 410.79
torch cuda    : 10.0.130 / is available
torch cudnn   : 7402 / is enabled

=== Hardware === 
nvidia gpus   : 1
torch devices : 1
  - gpu0      : 8116MB | GeForce GTX 1080

=== Environment === 
platform      : Linux-4.15.0-46-generic-x86_64-with-debian-stretch-sid
distro        : Ubuntu 16.04 xenial
conda env     : Unknown
python        : /home/mm/anaconda3/bin/python
sys.path      : /home/mm/anaconda3/lib/python36.zip
/home/mm/anaconda3/lib/python3.6
/home/mm/anaconda3/lib/python3.6/lib-dynload

/home/mm/anaconda3/lib/python3.6/site-packages
/home/mm/ssd2/ai/fastai
/home/mm/anaconda3/lib/python3.6/site-packages/IPython/extensions
/home/mm/.ipython
```

**Comment by sgugger** [comment]
Hi there! The training set has, by default, `drop_last=True`, which means the last batch with less than batch size samples is discarded. This is for stability in training when there are BatchNorm layers. To access the full training set you can either do:
```
data.train_dl = data.train_dl.new(drop_last=False)
```
which will replace the training dataloader with a new one that has `drop_last=False` or use `DatasetType.Fix`, which is the full training set (so no `drop_last`) with the same transforms as the validation set.
